hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/HoodieDatasetBulkInsertHelper.scala

```diff
@@ -155,17 +156,34 @@ object HoodieDatasetBulkInsertHelper
       val taskPartitionId = taskContextSupplier.getPartitionIdSupplier.get
       val taskId = taskContextSupplier.getStageIdSupplier.get.toLong
       val taskEpochId = taskContextSupplier.getAttemptIdSupplier.get
-      val writer = new BulkInsertDataInternalWriterHelper(
-        table,
-        writeConfig,
-        instantTime,
-        taskPartitionId,
-        taskId,
-        taskEpochId,
-        schema,
-        writeConfig.populateMetaFields,
-        arePartitionRecordsSorted,
-        shouldPreserveHoodieMetadata)
+
+      val writer = writeConfig.getIndexType match {
+        case HoodieIndex.IndexType.BUCKET if writeConfig.getBucketIndexEngineType
+          == BucketIndexEngineType.CONSISTENT_HASHING =>
+          new ConsistentBucketBulkInsertDataInternalWriterHelper(
+            table,
+            writeConfig,
+            instantTime,
+            taskPartitionId,
+            taskId,
+            taskEpochId,
+            schema,
+            writeConfig.populateMetaFields,
+            arePartitionRecordsSorted,
+            shouldPreserveHoodieMetadata)
+        case _ =>
```

Comment:
Should we support SIMPLE bucket index as well?

It uses `BucketBulkInsertDataInternalWriterHelper`.